src/overlays/actors/ovl_Boss_07/z_boss_07.c

```diff
@@ -242,500 +637,7210 @@ ActorProfile Boss_07_Profile = {
+    MAJORAS_INCARNATION_BODYPART_RIGHT_LEG_ROOT,  // MAJORAS_INCARNATION_LIMB_RIGHT_LEG_ROOT
+    MAJORAS_INCARNATION_BODYPART_RIGHT_THIGH,     // MAJORAS_INCARNATION_LIMB_RIGHT_THIGH
+    BODYPART_NONE,                                // MAJORAS_INCARNATION_LIMB_RIGHT_LOWER_LEG_ROOT
+    MAJORAS_INCARNATION_BODYPART_RIGHT_SHIN,      // MAJORAS_INCARNATION_LIMB_RIGHT_SHIN
+    MAJORAS_INCARNATION_BODYPART_RIGHT_FOOT,      // MAJORAS_INCARNATION_LIMB_RIGHT_FOOT
+    BODYPART_NONE,                                // MAJORAS_INCARNATION_LIMB_RIGHT_ARM_ROOT
+    MAJORAS_INCARNATION_BODYPART_RIGHT_UPPER_ARM, // MAJORAS_INCARNATION_LIMB_RIGHT_UPPER_ARM
+    BODYPART_NONE,                                // MAJORAS_INCARNATION_LIMB_RIGHT_LOWER_ARM_ROOT
+    MAJORAS_INCARNATION_BODYPART_RIGHT_FOREARM,   // MAJORAS_INCARNATION_LIMB_RIGHT_FOREARM
+    MAJORAS_INCARNATION_BODYPART_RIGHT_HAND,      // MAJORAS_INCARNATION_LIMB_RIGHT_HAND
+    MAJORAS_INCARNATION_BODYPART_LEFT_ARM_ROOT,   // MAJORAS_INCARNATION_LIMB_LEFT_ARM_ROOT
+    MAJORAS_INCARNATION_BODYPART_LEFT_UPPER_ARM,  // MAJORAS_INCARNATION_LIMB_LEFT_UPPER_ARM
+    BODYPART_NONE,                                // MAJORAS_INCARNATION_LIMB_LEFT_LOWER_ARM_ROOT
+    MAJORAS_INCARNATION_BODYPART_LEFT_FOREARM,    // MAJORAS_INCARNATION_LIMB_LEFT_FOREARM
+    MAJORAS_INCARNATION_BODYPART_LEFT_HAND,       // MAJORAS_INCARNATION_LIMB_LEFT_HAND
+    BODYPART_NONE,                                // MAJORAS_INCARNATION_LIMB_LEFT_LEG_ROOT
+    MAJORAS_INCARNATION_BODYPART_LEFT_THIGH,      // MAJORAS_INCARNATION_LIMB_LEFT_THIGH
+    BODYPART_NONE,                                // MAJORAS_INCARNATION_LIMB_LEFT_LOWER_LEG_ROOT
+    MAJORAS_INCARNATION_BODYPART_LEFT_SHIN,       // MAJORAS_INCARNATION_LIMB_LEFT_SHIN
+    MAJORAS_INCARNATION_BODYPART_LEFT_FOOT,       // MAJORAS_INCARNATION_LIMB_LEFT_FOOT
+    MAJORAS_INCARNATION_BODYPART_EYESTALK,        // MAJORAS_INCARNATION_LIMB_EYESTALK
+    BODYPART_NONE,                                // Doesn't correspond to a real limb on Majora's Incarnation
+    BODYPART_NONE,                                // Doesn't correspond to a real limb on Majora's Incarnation
+    BODYPART_NONE,                                // Doesn't correspond to a real limb on Majora's Incarnation
+    BODYPART_NONE,                                // Doesn't correspond to a real limb on Majora's Incarnation
+    BODYPART_NONE,                                // Doesn't correspond to a real limb on Majora's Incarnation
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Boss_07/func_80A07604.s")
+void Boss07_Incarnation_PostLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    Boss07* this = THIS;
+    Vec3f colliderPos;
+    s8 index;
+
+    if (limbIndex == MAJORAS_INCARNATION_LIMB_EYESTALK) {
+        Matrix_MultZero(&this->actor.focus.pos);
+    }
+
+    index = sIncarnationLimbToBodyParts[limbIndex];
```

Comment:
I deliberately didn't do this because we didn't do this in Odolwa. However, I just looked at Goht and we *do* in fact do it there. So let me just update both of these actors to be consistent with how Goht (and this suggestion) do it.